llvm/lib/Target/AIE/Utils/AIELoopUtils.cpp

```diff
@@ -86,6 +86,16 @@ getDedicatedFallThroughPreheader(const MachineBasicBlock &LoopBlock) {
   return Candidate;
 }
 
+SmallVector<const MachineBasicBlock *, 4>
+getInnermostLoopBlocks(const MachineLoopInfo &MLI) {
+  SmallVector<const MachineBasicBlock *, 4> LoopBBs;
+  for (const MachineLoop *ML : MLI) {
+    if (ML->isInnermost())
```

Comment:
How about considering only hardware loops here?

Reply:
I synchronized with Martien, what the `MBBs `are that we are looking for. The `MBBs` I am looking for are already checked with `isSingleMBBLoop `, so my function just returns a list of the found MBBs within a `MachineFunction `